src/bp-activity/bp-activity-functions.php

```diff
@@ -5568,3 +5568,212 @@ function bb_get_activity_hierarchy( $activity_id ) {
+	$activities_template->activity                  = $activity;
+	$activities_template->in_the_loop               = true;
+
+	if ( ! bb_activity_blog_post_acivity( $activity ) ) {
+		return $comments;
+	}
+
+	if ( ! bp_activity_can_comment() ) {
+		$comments = false;
+	}
+
+	return $comments;
+}
+add_filter( 'bb_activity_get_comments', 'bb_activity_has_comments', 10, 2 );
+
+/**
+ * Is it topic activity.
+ *
+ * @param object $activity Topic activity data.
+ *
+ * @since BuddyBoss 1.7.1
+ *
+ * @return bool
+ */
+function bb_activity_topic_acivity( $activity ) {
+	if ( empty( $activity ) ) {
+		return false;
+	}
+	
+	// When the activity type does not match with the topic.
+	if ( in_array( $activity->type, array( 'bbp_topic_create' ), true ) ) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * This function will give the topic id from topic activity.
+ *
+ * @param object $activity Topic activity data.
+ *
+ * @since BuddyBoss 1.7.1
+ *
+ * @return int
+ */
+function bb_activity_topic_id( $activity ) {
+	if ( empty( $activity ) ) {
+		return false;
+	}
+	
+	// When the activity type does not match with the topic.
+	if ( ! bb_activity_topic_acivity( $activity ) ) {
+		return false;
+	}
+
+	$topic_id = false;
+
+	// Set topic id when activity component is not groups.
+	if ( 'bbpress' === $activity->component ) {
+		// Set topic id when activity type topic.
+		$topic_id = $activity->item_id;
+	}
+
+	// Set topic id when activity component is groups.
+	if ( 'groups' === $activity->component ) {
+		// Set topic id when activity type topic.
+		$topic_id = $activity->secondary_item_id;
+	}
+
+	return $topic_id;
+}
+
+/**
+ * Is it topic reply activity.
+ *
+ * @param object $activity Reply activity data.
+ *
+ * @since BuddyBoss 1.7.1
+ *
+ * @return bool
+ */
+function bb_activity_topic_reply_acivity( $activity ) {
+	if ( empty( $activity ) ) {
+		return false;
+	}
+	
+	// When the activity type does not match with the topic.
+	if ( in_array( $activity->type, array( 'bbp_reply_create' ), true ) ) {
+		return true;
+	}
+
+	return false;
+}
+
+/**
+ * This function will give the reply topic id from reply activity.
+ *
+ * @param object $activity Reply activity data.
+ *
+ * @since BuddyBoss 1.7.1
+ *
+ * @return int
+ */
+function bb_activity_reply_topic_id( $activity ) {
+	if ( empty( $activity ) ) {
+		return false;
+	}
+
+	// When the activity type does not match with the topic or reply.
+	if ( ! bb_activity_topic_reply_acivity( $activity ) ) {
+		return false;
+	}
+
+	$topic_id = false;
+
+	// Set topic id when activity component is not groups.
+	if ( 'bbpress' === $activity->component ) {
+		// Set topic id when activity type reply.
+		$topic_id = bbp_get_reply_topic_id( $activity->item_id );
+	}
+
+	// Set topic id when activity component is groups.
+	if ( 'groups' === $activity->component ) {
+		// Set topic id when activity type reply.
+		$topic_id = bbp_get_reply_topic_id( $activity->secondary_item_id );
+	}
+
+	return $topic_id;
+}
```

Comment:
@KartikSuthar 
This is the helper function for API part. In further maybe we need it in web part. Because of which the function is place in web part.

Reply:
No need to remove after discuss with @KartikSuthar . Used in API